### PR TITLE
hw3

### PR DESCRIPTION
**Pull request by AnatoliyNik**

### CODE REVIEW

File: src/middleware/create-handler.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { NextFunction, Request, RequestHandler, Response } from "express";
+import { Model } from "mongoose";
+
```
Comment (vasylkivskyy) [inline src/middleware/create-handler.ts]: Read about differences about middlewares and controllers
From an architectural point of view, create-getAll-remove-update functions are controllers, so it is better to move them to controllers folder

File: src/db-models/genre.model.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        type: String,
+        required: true,
+        maxLength: 100,
+        trim: true,
+        lowercase: true,
+        validate: {
+            validator: async (name: string): Promise<boolean> => {
+                const count: number = await models[modelName.genre].countDocuments({name});
+
+                return !count;
```
Comment (vasylkivskyy) [inline src/db-models/genre.model.ts:16]: It is better to return 400 Status code, when validation failed, not 500

**Review by vasylkivskyy — CHANGES_REQUESTED** [review]

File: src/middleware/create-handler.ts (hunk 1 above)
Comment (AnatoliyNik) [reply src/middleware/create-handler.ts]: done

File: src/db-models/genre.model.ts (hunk 2 above)
Comment (AnatoliyNik) [reply src/db-models/genre.model.ts:16]: done

**Review by vasylkivskyy — APPROVED** [review]

**Pull request merged**